stack/bde_fdw_rds_stack.py

```diff
@@ -25,4 +26,38 @@ def __init__(  # type: ignore[no-untyped-def]
                 )
             )
 
-        aws_ec2.SubnetSelection(subnets=subnets)
+        vpc_subnets = aws_ec2.SubnetSelection(subnets=subnets)
+
+        # ----- Postgres with FDW RDS -----
+
+        # Generate ROOT credentials for postgresql rds with fdw and store it in secret manager
+        postgres_fdw_rds_root_cred_secret = aws_secretsmanager.Secret(
+            self,
+            "PostgresRDSCredentialSecretManagerRoot",
+            generate_secret_string=aws_secretsmanager.SecretStringGenerator(
+                exclude_characters="\"@/\\ '",
+                generate_string_key="password",
+                password_length=30,
+                secret_string_template='{"username": "postgres"}',
+            ),
+        )
+
+        # Create postgres rds instance with fdw
+        aws_rds.DatabaseInstance(
+            self,
+            "PostgresRDS_with_FDW_for_BDE_query",
+            database_name="PostgresRDS_with_FDW_for_BDE_query",
+            instance_type=aws_ec2.InstanceType.of(
+                getattr(aws_ec2.InstanceClass, rds_fdw_instance_type["class"]),
+                getattr(aws_ec2.InstanceSize, rds_fdw_instance_type["size"]),
+            ),
+            allocated_storage=10,
+            engine=aws_rds.DatabaseInstanceEngine.POSTGRES,
+            credentials=aws_rds.Credentials.from_secret(postgres_fdw_rds_root_cred_secret, "postgres"),
+            vpc=vpc,
+            vpc_subnets=vpc_subnets,
+            port=5432,
+            removal_policy=RemovalPolicy.DESTROY,
+            deletion_protection=False,  # TODO: Update accordingly
```

Comment:
`deletion_protection=environment=="prod"` might take care of it. At that point, though, you probably want to pull environments into constants, to consolidate magic strings.